### PR TITLE
meta2: fix memory leak

### PR DESCRIPTION
**Pull request by murlock**
##### SUMMARY

Fix memory leak

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
meta2

##### SDS VERSION
```
master
```

### CODE REVIEW

File: meta2v2/meta2_utils.c [hunk 1]
```diff
@@ -275,6 +275,8 @@ _load_chunk_quality(GHashTable *qualities, struct bean_PROPERTIES_s *prop)
 			g_hash_table_insert(qualities,
 					g_strdup(key->str + strlen(OIO_CHUNK_SYSMETA_PREFIX)), item);
 		} else {
+			g_free((char *)item->expected_slot);
+			g_free((char *)item->final_slot);
```
Comment (AymericDu) [inline meta2v2/meta2_utils.c]: If `meta2_json_fill_item_quality` returns an error, `expected_slot` and `final_slot` are not loaded: https://github.com/open-io/oio-sds/blob/master/meta2v2/meta2_utils_json_in.c#L297

File: meta2v2/meta2_utils.c [hunk 2]
```diff
@@ -2445,8 +2447,15 @@ m2db_check_content_quality(struct m2v2_sorted_content_s *sorted_content,
...
+
+	void _on_items(gpointer key UNUSED, struct oio_lb_selected_item_s *item, gpointer data UNUSED) {
+		g_free((char *) item->expected_slot);
+		g_free((char *) item->final_slot);
+	}
+
 	g_tree_foreach(sorted_content->metachunks,
 			(GTraverseFunc)_on_metachunk, NULL);
+	g_hash_table_foreach(chunk_items, (GHFunc)_on_items, NULL);
 	g_hash_table_destroy(chunk_items);
```
Comment (AymericDu) [inline meta2v2/meta2_utils.c]: You can use the fourth parameter of `g_hash_table_new_full` to free the values automatically: `https://github.com/open-io/oio-sds/blob/master/meta2v2/meta2_utils.c#L2403`

File: meta2v2/meta2_utils.c [hunk 3]
```diff
@@ -2445,8 +2447,15 @@ m2db_check_content_quality(struct m2v2_sorted_content_s *sorted_content,
...
 	}
+
+	void _on_items(gpointer key UNUSED, struct oio_lb_selected_item_s *item, gpointer data UNUSED) {
+		g_free((char *) item->expected_slot);
+		g_free((char *) item->final_slot);
+	}
+
 	g_tree_foreach(sorted_content->metachunks,
 			(GTraverseFunc)_on_metachunk, NULL);
+	g_hash_table_foreach(chunk_items, (GHFunc)_on_items, NULL);
```
Comment (fvennetier) [inline meta2v2/meta2_utils.c]: We should define `oio_lb_selected_item_free` and pass that to the hash table constructor.

File: meta2v2/meta2_utils.c (hunk 1 above)
Comment (fvennetier) [reply meta2v2/meta2_utils.c]: That is not a problem, `g_free` is NULL-safe.

**Pull request merged**